types/react-datepicker/index.d.ts

```diff
@@ -98,6 +98,7 @@ export interface ReactDatePickerProps<
     formatWeekDay?(day: string): React.ReactNode;
     formatWeekNumber?(date: Date): string | number;
     highlightDates?: Array<HighlightDates | Date> | undefined;
+    icon?: string | React.ReactNode;
```

Comment:
`ReactNode` includes `string` . https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L276:L287

When we pass React Component into `icon` prop, that seems to be expected `ReactElement` instead of `ReactNode` because that is judged by using `React.isValidElement()`.

https://github.com/Hacker0x01/react-datepicker/blob/main/src/calendar_icon.jsx#L7
https://react.dev/reference/react/isValidElement#isvalidelement

```suggestion
    icon?: string | React.ReactElement;
```

Reply:
Thanks for you input! I have updated the PR with the requested change.